src/bicep/mlz.bicep

```diff
@@ -1,30 +1,457 @@
 targetScope = 'subscription'
 
+/*
+
+  PARAMETERS
+
+  Here are all the parameters a user can override.
+
+  These are the required parameters that Mission LZ does not provide a default for:
+    - resourcePrefix
+
+*/
+
+// REQUIRED PARAMETERS
+
+@minLength(3)
+@maxLength(10)
+@description('A prefix, 3-10 alphanumeric characters without whitespace, used to prefix resources and generate uniqueness for resources with globally unique naming requirements like Storage Accounts and Log Analytics Workspaces')
+param resourcePrefix string
+
+@description('The subscription ID for the Hub Network and resources. It defaults to the deployment subscription.')
+param hubSubscriptionId string = subscription().subscriptionId
+
+@description('The subscription ID for the Identity Network and resources. It defaults to the deployment subscription.')
+param identitySubscriptionId string = subscription().subscriptionId
+
+@description('The subscription ID for the Operations Network and resources. It defaults to the deployment subscription.')
+param operationsSubscriptionId string = subscription().subscriptionId
+
+@description('The subscription ID for the Shared Services Network and resources. It defaults to the deployment subscription.')
+param sharedServicesSubscriptionId string = subscription().subscriptionId
+
+@description('The region to deploy resources into. It defaults to the deployment location.')
+param location string = deployment().location
+
+// RESOURCE NAMING PARAMETERS
+
+@minLength(3)
+@maxLength(6)
+@description('A suffix, 3 to 6 characters in length, to append to resource names (e.g. "dev", "test", "prod", "mlz"). It defaults to "mlz".')
+param resourceSuffix string = 'mlz'
```

Comment:
Should this be after `resourcePrefix`? It's a complementary parameter to `resourcePrefix`.

Reply:
Moved it closer to `resourcePrefix` since it's complementary and makes it easier to discover 👍